lib/position.rb

```diff
@@ -0,0 +1,13 @@
+class Position
```

Comment:
This is a value in mathematics and therefore may want to be modelled as a value object which tend to be immutable (only create new objects, don't modify the current value's state). This impacts the `up` method. What might we want to do instead in `up`?